internal/plugincommon/discovery/processor.go

```diff
@@ -114,6 +113,72 @@ func (cdp *ContractDiscoveryProcessor) ValidateObservation(
 	return nil
 }
 
+type aggObs struct {
+	onrampAddrs       map[cciptypes.ChainSelector][][]byte
+	feeQuoterAddrs    map[cciptypes.ChainSelector][][]byte
+	nonceManagerAddrs [][]byte
```

Comment:
won't nonceManager and routers be easier to deal with based on ChainSelectors as well?

Reply:
updated.